parser_library/src/processing/instruction_sets/ca_processor.cpp

```diff
@@ -693,3 +697,179 @@ void ca_processor::process_MHELP(const processing::resolved_statement& stmt)
+
+        using enum context::SET_t_enum;
+        if (const auto type = evaluated.type(); type != A_TYPE && type != B_TYPE)
+        {
+            add_diagnostic(diagnostic_op::error_E013("SETAF", ca_op->operand_range));
+            return;
+        }
+        args.push_back(evaluated.access_a());
+    }
+
+    if (ops.back()->type == semantics::operand_type::EMPTY) // last empty operand ignored
+        args.pop_back();
+
+    external_function_args func_args(args);
+    const auto& typed_arg = *func_args.arithmetic();
+
+    (*func)(func_args);
+
+    if (auto& message = func_args.message(); message)
+        process_external_message(*message, func_name, stmt.instruction_ref().field_range);
+
+    auto& val = set_symbol->access_set_symbol<context::A_t>()->reserve_value(index);
+    val = typed_arg.result;
+}
+
+void ca_processor::process_SETCF(const resolved_statement& stmt)
+{
+    auto [set_symbol, name, index] = get_SET_symbol<context::C_t>(stmt);
+
+    if (!set_symbol)
+        return;
+
+    const auto& ops = stmt.operands_ref().value;
+    if (ops.empty())
+    {
+        add_diagnostic(diagnostic_op::error_E022("SETCF", stmt.instruction_ref().field_range));
+        return;
+    }
+
+    const auto [func, func_name] = find_external_function(*ops.front());
+    if (!func)
+        return;
+
+    std::vector<context::C_t> args;
+    args.reserve(ops.size() - 1);
+    for (const auto& op : ops | std::views::drop(1))
+    {
+        if (op->type == semantics::operand_type::EMPTY)
+        {
+            args.emplace_back();
+            continue;
+        }
+
+        auto ca_op = op->access_ca();
+        assert(ca_op);
+
+        if (ca_op->kind != semantics::ca_kind::VAR && ca_op->kind != semantics::ca_kind::EXPR)
+        {
+            add_diagnostic(diagnostic_op::error_E012("SETCF", ca_op->operand_range));
+            return;
+        }
+
+        auto evaluated = ca_op->access_expr()->expression->evaluate(eval_ctx);
+
+        if (const auto type = evaluated.type(); type != context::SET_t_enum::C_TYPE)
+        {
+            add_diagnostic(diagnostic_op::error_E013("SETCF", ca_op->operand_range));
+            return;
+        }
+        args.push_back(std::move(evaluated.access_c()));
+    }
+
+    if (ops.back()->type == semantics::operand_type::EMPTY) // last empty operand ignored
+        args.pop_back();
+
+    const std::vector<std::string_view> views(args.begin(), args.end());
+    external_function_args func_args(views);
+    const auto& typed_arg = *func_args.character();
+
+    (*func)(func_args);
+
+    if (auto& message = func_args.message(); message)
+        process_external_message(*message, func_name, stmt.instruction_ref().field_range);
+
+    auto& val = set_symbol->access_set_symbol<context::C_t>()->reserve_value(index);
+
+    static constexpr size_t external_string_limit = 1024;
+
+    auto res = utils::replace_non_utf8_chars(std::string_view(typed_arg.result).substr(0, 4 * external_string_limit));
```

Comment:
Does it make sense to issue a warning/info message that the text is being truncated?